http/peer-id-auth.md

```diff
@@ -0,0 +1,194 @@
+| challenge-server | The random base64 encoded value the client generates to challenge the server to prove its identity                                                                                                                                       |
+| challenge-client | The random base64 encoded value the server generates to challenge the client to prove its identity                                                                                                                                       |
+| sig              | A base64 encoded signature.                                                                                                                                                                                                              |
+| peer-id          | The peer id of the node that set this parameter. Encoding defined by the [Peer ID spec].                                                                                                                                                 |
+| public-key       | A base64 encoded value of peer's public key. The key itself is encoded per the [Peer ID spec].                                                                                                                                           |
+| opaque           | An base64 encoded opaque to the client blob generated by the server. If a client receives this it must return it. A server may use this to authenticate statelessly. For example, it could store the challenge-client and a expiry time. |
+
+Params are encoded per [RFC 9110 auth-param's ABNF](https://datatracker.ietf.org/doc/html/rfc9110#name-collected-abnf). Generally it'll be something like: `hostname="example.com", challenge-server=base64EncodedVal`
+
+## Signing
+
+Signatures sign some set of parameters. The parameters are sorted
+alphabetically, prepended with a varint length prefix, and concatenated together
+to form the data to be signed. The signing algorithm is defined by the key type
+used. Refer to the [Peer ID
+spec] for
+specifics on the signing algorithm. The set of parameters is prefixed with the auth scheme "libp2p-PeerID"
+
+As an example, if we wanted to sign the parameters `hostname="example.com",
+challenge-client=base64String` we would first structure the parameters as a byte
+slice containing:
+```
+libp2p-PeerID<varintprefix>challenge-client=<base64String><varintprefix>hostname="example.com"
+```
+
+Then sign the resulting byte slice. See the test vectors below for a
+examples.
+
+
+## Base64 Encoding
+
+The base64 encoding follows Base 64 Encoding with URL and Filename Safe Alphabet
+from [RFC 4648](https://datatracker.ietf.org/doc/html/rfc4648#section-5). The
+reason this is not a multibase is to aid clients or servers who can not or
+prefer not to import a multibase dependency.
```

Comment:
I think it's still better to keep this simple. But let me be the first to point out that users of this auth scheme implicitly rely on multibase when they handle peer id decoding.

Reply:
I think this is fine. Multibase is useful if you want to share the bytes with other people. We have no such need here.